vividus-plugin-rest-api/src/test/java/org/vividus/bdd/steps/api/HttpRequestStepsTests.java

```diff
@@ -149,6 +157,35 @@ void testSetCustomRequestConfigNoField()
         assertEquals("nonExistentField", exception.getMessage());
     }
 
+    @Test
+    void testWaitForResponseCode()
+    {
+        SubSteps stepsToExecute = mock(SubSteps.class);
+        when(httpTestContext.getResponse()).thenReturn(new HttpResponse());
+        httpRequestSteps.waitForResponseCode(RESPONSE_CODE, Duration.ofSeconds(10), RETRY_TIMES, stepsToExecute);
+        verify(stepsToExecute, times(RETRY_TIMES)).execute(Optional.empty());
+    }
+
+    @Test
+    void testWaitForResponseCodeWhenResponseIsNull()
+    {
+        SubSteps stepsToExecute = mock(SubSteps.class);
+        when(httpTestContext.getResponse()).thenReturn(null);
+        httpRequestSteps.waitForResponseCode(RESPONSE_CODE, Duration.ofSeconds(10), RETRY_TIMES, stepsToExecute);
+        verify(stepsToExecute, times(RETRY_TIMES)).execute(Optional.empty());
+    }
```

Comment:
this test and the previous one can be combined and data-driven approach can be applied: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource